readthedocs/builds/forms.py

```diff
@@ -33,3 +39,12 @@ def save(self, *args, **kwargs):
         obj = super(VersionForm, self).save(*args, **kwargs)
         if obj.active and not obj.built and not obj.uploaded:
             trigger_build(project=obj.project, version=obj)
+
+    def clean(self):
+        cleaned_data = super(VersionForm, self).clean()
+        if self.instance.pk is not None:  # new instance only
+            if self.instance.active is True and cleaned_data['active'] is False:
```

Comment:
Django will actually change the `self.instance` attributes before it calls the `save` method somewhere in the full clean cycle. That has bitten me a few times in the past.

Therefore I think the best place to grab old, unsaved values is in the `__init__` method and save those original values into a form attribute. Then using this to test if the value has changed.

In this particular case you could use the `self.changed_data` attribute: https://docs.djangoproject.com/en/1.8/ref/forms/api/#django.forms.Form.changed_data